docs/design/resource_controller.md

```diff
@@ -0,0 +1,269 @@
+type ResourceControllerGroup struct {
+  GroupBy   ResourceControllerGroupBy  `json:"groupBy,omitempty"`
+  RuleType  ResourceControllerRuleType `json:"ruleType,omitempty"`
+  Resources ResourceList               `json:"resources,omitempty"`
+}
+
+type ResourceControllerSpec struct {
+  // Allowed represents the available resources allowed in a quota
+  Allowed []ResourceControllerGroup `json:"allowed,omitempty"`
+}
+
+type ResourceControllerStatus struct {
+  // Allowed represents the available resources allowed across scopes
+  Allowed []ResourceControllerGroup `json:"allowed,omitempty"`
+  // Allocated represents the allocated resources leveraged across scopes
+  Allocated []ResourceControllerGroup `json:"allocated,omitempty"`
+}
+
+// ResourceControllerList is a collection of resource controllers.
+type ResourceControllerList struct {
+  TypeMeta `json:",inline"`
+  ListMeta `json:"metadata,omitempty"`
+  Items    []ResourceController `json:"items"`
+}
+```
+
+## AdmissionControl plugin: ResourceLimits
+
+The **ResourceLimits** plug-in introspects all incoming admission requests. 
+
+It makes decisions by evaluating the incoming object against current status and the allowed resources enumerated on **ResourceController**.
+
+The following are the set of proposed rules by group.
+
+**GroupBy: Container**
+
+| RuleType | ResourceName | Description |
+| ------ | -------- | -------- |
+| Max | CPU | Maximum amount of CPU per container |
+| Max | Memory | Maximum amount of memory per container |
+| Min | CPU | Minimum amount of CPU per container |
+| Min | Memory | Minimum amount of memory per container |
+
+**GroupBy: Pod**
+
+| RuleType | ResourceName | Description |
+| ------ | -------- | -------- |
+| Max | CPU | Maximum amount of CPU per pod |
+| Max | Memory | Maximum amount of memory per pod |
+| Min | CPU | Minimum amount of CPU per pod |
+| Min | Memory | Minimum amount of memory per pod |
+
+**GroupBy: ReplicationController**
+
+| RuleType | ResourceName | Description |
+| ------ | -------- | -------- |
+| Max | Replicas | Maximum amount of replicas  per replication controller |
+
+**GroupBy: Namespace**
+
+| RuleType | ResourceName | Description |
+| ------ | -------- | -------- |
+| Max | CPU | Maximum amount of CPU per namespace |
+| Max | Memory | Maximum amount of memory per namespace |
+| Max | Pods | Maximum amount of pods per namespace |
+| Max | ReplicationControllers | Maximum amount of replication controllers per namespace |
+| Max | Services | Maximum amount of services per namespace |
+
+If the incoming object would cause a violation of the enumerated constraints, the request is denied with a set of
+messages explaining what constraints were the source of the denial.
+
+If a constraint is not enumerated by a **ResourceController** it is not tracked.
+
+If a constraint spans resources, for example, it tracks the total number of some **kind** in a **namespace**,
```

Comment:
I don't understand what you mean by *If a constraint spans resources*.  Do you mean it spans multiple pods? If so, then compute-resource and rest-resource need to be disambiguated in this document.